lua/ide/components/bufferlist/component.lua

```diff
@@ -93,6 +96,14 @@ BufferListComponent.new = function(name, config)
             end,
             group = vim.api.nvim_create_augroup("NvimIdeBufferlist", { clear = true }),
         })
+        vim.api.nvim_create_autocmd({ "BufEnter" }, {
+            callback = function(args)
+                if libbuf.is_regular_buffer(args.buf) then
```

Comment:
here we probably want to check the same condition as `libbuf.get_listed_bufs()` which is `vim.bo[bufnr].buflisted and vim.api.nvim_buf_is_loaded(bufnr) and vim.api.nvim_buf_get_name(bufnr) ~= ""`.

This check can be extracted from `libbuf.get_listed_bufs()` into a separate utility function that can be used here as well.